packages/ipfs/src/core/ipns/routing/utils.js

```diff
@@ -3,14 +3,34 @@
 const multibase = require('multibase')
 const ipns = require('ipns')
 
+/**
+ * @typedef {Object} Record
+ * @todo Figure out actual structure
+ */
+
 module.exports = {
+  /**
+   * @param {Buffer} buf
+   * @returns {string}
+   */
   encodeBase32: (buf) => {
     const m = multibase.encode('base32', buf).slice(1) // slice off multibase codec
 
     return m.toString().toUpperCase() // should be uppercase for interop with go
   },
   validator: {
-    func: (key, record, cb) => ipns.validator.validate(record, key, cb)
```

Comment:
validator doesn't take `cb` (anymore) so I changed this to reflect current state.